quicklens/src/main/scala/com/softwaremill/quicklens/package.scala

```diff
@@ -115,12 +115,12 @@ package object quicklens {
     def each(fa: F[K, T])(f: T => T): F[K, T]
   }
 
-  implicit def mapQuicklensFunctor[K, T] = new QuicklensMapAtFunctor[Map, K, T] {
-    override def at(fa: Map[K, T], key: K)(f: T => T) = {
-      fa.updated(key, f(fa(key)))
+  implicit def mapQuicklensFunctor[M[KT, TT] <: Map[KT, TT], K, T] = new QuicklensMapAtFunctor[M, K, T] {
+    override def at(fa: M[K, T], key: K)(f: T => T) = {
+      fa.updated(key, f(fa(key))).asInstanceOf[M[K, T]]
     }
-    override def each(fa: Map[K, T])(f: (T) => T) = {
-      fa.mapValues(f)
+    override def each(fa: M[K, T])(f: (T) => T) = {
+      fa.mapValues(f).asInstanceOf[M[K, T]]
```

Comment:
When I tried to provide tests for HashMap (and ListMap), I have found while `at` works fine, modification using `each` fails the test, as the cast cannot be done (mapValues is not preserving the collection type). I suggest not merging until I fix this.